torchtext/datasets/wikitext103.py

```diff
@@ -28,11 +29,23 @@
 @_add_docstring_header(num_lines=NUM_LINES)
 @_create_dataset_directory(dataset_name=DATASET_NAME)
 @_wrap_split_argument(('train', 'valid', 'test'))
-def WikiText103(root, split):
-    dataset_tar = download_from_url(URL, root=root, hash_value=MD5, hash_type='md5')
-    extracted_files = extract_archive(dataset_tar)
-
-    path = _find_match(split, extracted_files)
-    logging.info('Creating {} data'.format(split))
-    return _RawTextIterableDataset(DATASET_NAME,
-                                   NUM_LINES[split], _read_text_iterator(path))
+def WikiText103(root: str, split: Union[Tuple[str], str]):
+    if not is_module_available("torchdata"):
+        raise ModuleNotFoundError("Package `torchdata` not found. Please install following instructions at `https://github.com/pytorch/data`")
+    url_dp = IterableWrapper([URL])
+    # cache data on-disk
+    filepath_fn = functools.partial(lambda x: os.path.join(root, os.path.basename(x)))
```

Comment:
Should we add lambda function directly to filepath_fn since partial doesn't help solve the pickling issue? Also I am curious if we should be strict about making the functional pickable?

Reply:
As decided, we will move these lambda functions later in a consolidated effort. Hence resolving this comment for now.